sdks/go/cmd/starcgen/starcgen.go

```diff
@@ -57,6 +57,13 @@ func Generate(w io.Writer, filename, pkg string, ids []string, fset *token.FileS
 
 	e.Print("*/\n")
 	data := e.Generate(filename)
+	if err := write(w, []byte(license)); err != nil {
+		return err
```

Comment:
Do generated files need to have a Apache license?

Reply:
In principle: probably not. 
In practice: the RAT check will likely complain if we don't, when trying to check in the files.

Note: We need to check in the files, since the design of `go generate` requires it be invoked manually, and the results are intended to be checked in.
Go says: `go generate` is for library authors, not library users.
Similarly, the protos are generated code, but we need to check in their results. (They don't have the license though, but they may pre-date the RAT check).

Reply:
Ack. But this is strange for the end users of the tool. They do not necessarily need to want to use a this license. 

Perhaps a TODO for the future to see if we can resolve this with the RAT tool.

Reply:
Given it's an Apache tool generating the code, it's not a huge stretch that it should include the Apache license.

And a user that takes particular umbrage can always use the starcgenx package to write their own tooling to generate the code. 
I would happily accept a patch (or spend the 2 minutes necessary to write the code) to flag the license inclusion lines, if a user requested it. I'd rather not speculate on possible usecases, and generalizing/flagging before there's a concrete need.